tests/serac_odes.cpp

```diff
@@ -0,0 +1,645 @@
+  if (m == serac::TimestepMethod::ForwardEuler) return std::string("ForwardEuler");
+  if (m == serac::TimestepMethod::RK2) return std::string("RK2");
+  if (m == serac::TimestepMethod::RK3SSP) return std::string("RK3SSP");
+  if (m == serac::TimestepMethod::RK4) return std::string("RK4");
+  if (m == serac::TimestepMethod::GeneralizedAlpha) return std::string("GeneralizedAlpha");
+  if (m == serac::TimestepMethod::ImplicitMidpoint) return std::string("ImplicitMidpoint");
+  if (m == serac::TimestepMethod::SDIRK23) return std::string("SDIRK23");
+  if (m == serac::TimestepMethod::SDIRK34) return std::string("SDIRK34");
+
+  // for second order odes
+  if (m == serac::TimestepMethod::Newmark) return std::string("Newmark");
+  if (m == serac::TimestepMethod::HHTAlpha) return std::string("HHTAlpha");
+  if (m == serac::TimestepMethod::WBZAlpha) return std::string("WBZAlpha");
+  if (m == serac::TimestepMethod::AverageAcceleration) return std::string("AverageAcceleration");
+  if (m == serac::TimestepMethod::LinearAcceleration) return std::string("LinearAcceleration");
+  if (m == serac::TimestepMethod::CentralDifference) return std::string("CentralDifference");
+  if (m == serac::TimestepMethod::FoxGoodwin) return std::string("FoxGoodwin");
+  return "unknown";
+}
+
+std::string to_string(serac::DirichletEnforcementMethod m)
+{
+  // for first order odes
+  if (m == serac::DirichletEnforcementMethod::DirectControl) return std::string("direct control");
+  if (m == serac::DirichletEnforcementMethod::RateControl) return std::string("rate control");
+  if (m == serac::DirichletEnforcementMethod::FullControl) return std::string("full control");
+  return "unknown";
+}
+
+int order_of_convergence(serac::TimestepMethod m)
+{
+  // for first order odes
+  if (m == serac::TimestepMethod::BackwardEuler) return 1;
+  if (m == serac::TimestepMethod::SDIRK33) return 3;  // ?
+  if (m == serac::TimestepMethod::ForwardEuler) return 1;
+  if (m == serac::TimestepMethod::RK2) return 2;     // ?
+  if (m == serac::TimestepMethod::RK3SSP) return 3;  // ?
+  if (m == serac::TimestepMethod::RK4) return 4;
+  if (m == serac::TimestepMethod::GeneralizedAlpha) return 2;  // ?
+  if (m == serac::TimestepMethod::ImplicitMidpoint) return 2;
+  if (m == serac::TimestepMethod::SDIRK23) return 2;
+  if (m == serac::TimestepMethod::SDIRK34) return 3;
+
+  // for second order odes
+  if (m == serac::TimestepMethod::Newmark) return 2;
+  if (m == serac::TimestepMethod::HHTAlpha) return 2;
+  if (m == serac::TimestepMethod::WBZAlpha) return 2;             // ?
+  if (m == serac::TimestepMethod::AverageAcceleration) return 2;  // ?
+  if (m == serac::TimestepMethod::LinearAcceleration) return 2;   // ?
+  if (m == serac::TimestepMethod::CentralDifference) return 2;    // ?
+  if (m == serac::TimestepMethod::FoxGoodwin) return 2;           // ?
+
+  return -1;
+}
+
+int which_kind_of_ode(serac::TimestepMethod m)
```

Comment:
Maybe `order_of_ode_solver`?

Reply:
I'm trying to avoid "order" since it's ambiguous (order of convergence or first/second order), but `kind` isn't great either! I'll leave a descriptive comment for now.

Reply:
Agreed. I'll bring this up in a meeting and see if anyone has ideas.